app/controllers/dashboard_controller.rb

```diff
@@ -3,7 +3,14 @@ def index
     flash.now[:alert] = "OnDemand requires a newer version of the browser you are using. Current browser requirements include IE Edge, Firefox 19+, Chrome 34+, Safari 8+." unless view_context.browser.modern?
     flash.now[:alert] = "OnDemand is not yet optimized for mobile use." if view_context.browser.device.mobile?
 
-    @motd_file = MotdFile.new
+    @motd_file = case ENV['MOTD_FORMAT']
+    when 'osc'
+      MotdFile::MotdFormatterOsc.new
+    when 'markdown'
+      MotdFile::MotdFormatterMarkdown.new
+    else
+      MotdFile::MotdFormatterPlainText.new
+    end
```

Comment:
Instead of inheritance, I'd consider:

```
file = MotdFile.new(ENV["MOTD_PATH")
formatted_file = MotdFile::MotdFormatterOsc.new(file)
```

i.e. instead of 

```ruby
class MotdFormatterPlainText < MotdFile
  def initialize
    super
    @content = motd_content
  end

  def to_partial_path
    "dashboard/motd_text"
  end
end
```

something like

```ruby
class MotdFormatterPlainText
  def initialize(motd_file)
    @content = motd_file.content
  end

  def to_partial_path
    "dashboard/motd_text"
  end
end
```